Bio/Seq.py

```diff
@@ -95,26 +95,41 @@ def __init__(self, data):
             self._data = bytes(data)
         elif isinstance(data, str):
             self._data = bytes(data, encoding="ASCII")
+        elif self._check_bytes_like(data):
+            self._data = data
```

Comment:
I would suggest added a comment here, something like:

``# e.g. the TwoBitSequence objects which can act like a view onto a sub-sequence``

Reply:
We may want to define a abstract base class (as a separate PR) to specify what exactly we mean by a bytes-like object here. Python itself has a similar concept (anything that implements the buffer interface for bytes).

Reply:
Sounds sensible.